src/content/en/fundamentals/primers/shadowdom/index.markdown

```diff
@@ -689,21 +689,108 @@ If the shadow tree is open, calling `event.composedPath()` will return an array
 Custom DOM events which are fired on internal nodes in a shadow tree do not bubble
 out of the shadow boundary unless the event is created using the `composed: true` flag:
 
-    // Inside <fancy-tab> custom element class definition:
-    selectTab() {
-      const tabs = this.shadowRoot.querySelector('#tabs');
-      tabs.dispatchEvent(new Event('tab-select', {bubbles: true, composed: true}));
-    }
+{% highlight javascript %}
+// Inside <fancy-tab> custom element class definition:
+selectTab() {
+  const tabs = this.shadowRoot.querySelector('#tabs');
+  tabs.dispatchEvent(new Event('tab-select', {bubbles: true, composed: true}));
+}
+{% endhighlight %}
 
 If `composed: false` (default), consumers won't be able to listen for the event outside of your shadow root.
 
-    <fancy-tabs></fancy-tabs>
-    <script>
-      const tabs = document.querySelector('fancy-tabs');
-      tabs.addEventListener('tab-select', e => {
-        // won't fire if `tab-select` wasn't created with `composed: true`.
-      });
-    </script>
+{% highlight html %}
+<fancy-tabs></fancy-tabs>
+<script>
+  const tabs = document.querySelector('fancy-tabs');
+  tabs.addEventListener('tab-select', e => {
+    // won't fire if `tab-select` wasn't created with `composed: true`.
+  });
+</script>
+{% endhighlight %}
+
+### Handling focus {#focus}
+
+If you recall from [shadow DOM's event model](#events), events that are fired inside shadow DOM are adjusted to look like they come from the hosting element. For example, let's say you click an `<input>` inside a shadow root:
+
+{% highlight html %}
+<x-focus>
+  #shadow-root
+    <input type="text" placeholder="Input inside shadow dom">
+{% endhighlight %}
+
+ The `focus` event will look like it came from `<x-focus>`, not the `<input>`. Similarly, `document.activeElement` will be `<x-focus>`. If the shadow root was created with `mode:'open'` (see [closed mode](#closed)), you'll also be able access the internal node that gained focus:
+
+    document.activeElement.shadowRoot.activeElement // only works with open mode.
+
+Another option is `delegatesFocus: true`, which expands the focus behavior of element's within a shadow tree:
+
+- If you click on a node inside shadow DOM and the node is not a focusable area, the first focusable area becomes focused.
```

Comment:
This actually felt right to me but who knows. It's also opt-in so maybe the use case is the "widget", where you have focusable items inside the shadow dom. When you tab to the widget, you should be focused on the first focusable item.

cc @TakayoshiKochi